### ISSUE TITLE
Support Windows builds in release/2.1 up to Core-Setup

### ISSUE THREAD

**Issue by dagood** [dotnet/source-build#898]
This issue is to support Windows builds up to Core-Setup (Microsoft.NETCore.App), add it to AzDO CI, and flow the work through to `release/2.2` and `master`. We have non-Windows-only change debt, and want to get ahead of it before the more dramatic build flow changes coming soon.

`master` currently has a Windows build, but it is a special case causing difficulties when auto-merging and upgrading.

A few things that can be built on top of this, but aren't included:
* Build the SDK. We know we have some issues to work through in the submodules before we can do this.
* https://github.com/dotnet/source-build/issues/904 Smoke-test support. (Requires a full SDK/CLI build, right now. Maybe we could work around it to use the tooling SDK with source-built Runtime.)
* Building a tarball that can build (offline) on Windows. This doesn't address any known real-world scenario. The only thing I can think of is it would potentially be a convenience for Microsoft devs to work with the offline build flow while on Windows, if they aren't used to Linux.

**Comment by dagood** [comment]
Closing, merged into 2.1. New issues: #904 (smoke-test) #907 (prebuilt baseline)
